src/cpp/src/llm_pipeline.cpp

```diff
@@ -65,6 +68,22 @@ std::pair<EncodedResults, int32_t> beam_search(
 );
 
 class StatefulLLMPipeline final : public LLMPipelineImplBase {
+private:
+    void slice_matmul_statefull_model(std::shared_ptr<ov::Model> model) {
+        auto last_node = model->output(0).get_node()->input_value(0).get_node();
+        if (auto matmul = dynamic_cast<ov::op::v0::MatMul*>(last_node)) {
+            auto shape = matmul->input(0).get_partial_shape();
```

Comment:
It seems you don't access `StatefulLLMPipeline`'s state. `slice_matmul_statefull_model()` should be standalone function.